### PR TITLE
[PR] Implement `wp wsuwp site create` command

### PR DESCRIPTION
**Pull request by jeremyfelt**
This allows us to create sites via WP-CLI that match the custom new site screen we've built for WSUWP.

`wp wsuwp site create web.wsu.edu/sub-site jeremy.felt@wsu.edu "Web Sub Site" 3` creates a new site with the web.wsu.edu/sub-site URL on network ID 3.

### CODE REVIEW

File: commands/wsuwp-cli-site.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	 * : The network ID of the first admin for the site.
+	 *
+	 * <site-name>
+	 * : The name of the site.
+	 *
+	 * <network-id>
+	 * : The ID of the network the site should be created under.
+	 *
+	 * [--porcelain]
+	 * : Output just the new user id.
```
Comment (philcable) [inline commands/wsuwp-cli-site.php]: s/user/site :)

File: commands/wsuwp-cli-site.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		$existing = get_sites( array(
+			'domain' => $new_site->domain,
+			'path' => $new_site->path,
+		) );
+
+		if ( 0 !== count( $existing ) ) {
+			WP_CLI::error( 'A site with this domain and path combination already exists.' );
+		}
+
+
```
Comment (philcable) [inline commands/wsuwp-cli-site.php]: Extra empty line here.

File: commands/wsuwp-cli-site.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		$user_id = email_exists( $new_site->nid );
+
+		if ( ! $user_id ) { // Create a new user with a random password
+			WP_CLI::error( 'The user does not exist.' );
+		}
+
```
Comment (philcable) [inline commands/wsuwp-cli-site.php:90]: Should we check that the given network exists, too, for good measure?

File: commands/wsuwp-cli-site.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		// Clear any stale cache related to this domain and path request. See sunrise.
+		wp_cache_delete( $new_site->domain . $new_site->path, 'wsuwp:site' );
+
+		$content_mail = sprintf( __( 'New site created by %1$s
+
+Address: %2$s
+Name: %3$s' ), wp_get_current_user()->user_login , get_site_url( $id ), wp_unslash( $new_site->name ) );
+		wp_mail( get_site_option( 'admin_email' ), sprintf( __( '[%s] New Site Created' ), get_current_site()->site_name ), $content_mail, 'From: "Site Admin" <' . get_site_option( 'admin_email' ) . '>' );
```
Comment (philcable) [inline commands/wsuwp-cli-site.php]: I'm going to show my WP CLI - or maybe my network of networks - ignorance here: is there a "current network" context (for `get_site_option()`) when this command runs?

File: commands/wsuwp-cli-site.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+class Command {
+	/**
+	 * Create a new site.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <site-url>
+	 * : The domain and path of the new site.
+	 *
+	 * <wsu-nid>
```
Comment (philcable) [inline commands/wsuwp-cli-site.php]: Here and throughout, it looks like `wsu-nid` should be `user-email`.

File: features/wsuwp-site.feature [hunk 6]
```diff
@@ -0,0 +1,56 @@
+Feature: Test the wsuwp site command
+
+  Scenario: WSUWP site command exists
+    Given a WSUWP Platform install
+
+    When I run `wp wsuwp site`
+    Then STDOUT should contain:
+      """
+      usage: wp wsuwp site create <site-url> <wsu-nid> <site-name> <network-id> [--porcelain]
```
Comment (philcable) [inline features/wsuwp-site.feature]: s/wsu-nid/user-email here too.

**Review by philcable — COMMENTED** [review]
Looking good! I left a few inline comments and questions.

File: commands/wsuwp-cli-site.php (hunk 4 above)
Comment (jeremyfelt) [reply commands/wsuwp-cli-site.php]: Yeah, this technically runs in the "current network" context. It's probably a good idea to change that to use the network that we're creating the site for!

File: commands/wsuwp-cli-site.php (hunk 3 above)
Comment (jeremyfelt) [reply commands/wsuwp-cli-site.php:90]: We should. Good call 👍

**Comment by jeremyfelt** [comment]
@philcable #reviewmerge?

**Review by philcable — APPROVED** [review]
👍  Looks good!

**Pull request merged**